jit.c

```diff
@@ -932,8 +932,8 @@ static void rv_jit_clear(struct riscv_t *rv)
 static void block_map_free(struct block_map_t *map)
 {
     assert(map->map);
-    for (uint32_t i = 0; i < map->size; i++)
-        free(map->map[i]);
+    free(map->map);
```

Comment:
Why did you make this change?

Reply:
Since block will be used in latter, so I just free map instead of block in map.
May be I should write another function for different case.

Reply:
If you change the semantics of a certain function, you need explicit motivation and explanations.

Reply:
`block_map_free` should free block map, if I want free blocks that I should call `block_map_clear`.

For this case `block_map_enlarge`, the blocks will be used in latter, so I just translate blocks and free block map to avoid segement falut.

If I want free block map safely,  `rv_jit_free` should write like follow.
```
void rv_jit_free(struct riscv_jit_t *jit)
{
    if (jit->block_map) {
        block_map_clear(jit->block_map);
        block_map_free(jit->block_map);
        jit->block_map->map = NULL;
    }
...
}
```

Reply:
Write down the corresponding comments and check the usage to adopt.